src/client/ui/panels/PropertiesPanelContainer.js

```diff
@@ -119,51 +121,96 @@ class PropertiesPanelContainer extends Component {
-          confirmButtonLabel: "Save"
+      this.props.showDialog(FileDialog, {
+        filters: [component.fileExtension],
+        extension: component.fileExtension,
+        title: "Save material as...",
+        confirmButtonLabel: "Save",
+        onConfirm: async src => {
+          let saved = false;
+
+          try {
+            setTimeout(() => {
+              if (saved) return;
+              this.showDialog(ProgressDialog, {
+                title: "Saving Material",
+                message: "Saving material..."
+              });
+            }, 100);
+
+            component.src = src;
+            component.shouldSave = true;
+            await this.props.project.writeJSON(component.src, component.props);
+            component.modified = false;
+            this.props.editor.signals.objectChanged.dispatch(this.state.object);
+            this.hideDialog();
+          } catch (e) {
+            this.showDialog(ErrorDialog, {
+              title: "Error saving material",
+              message: e.message
+            });
+          } finally {
+            saved = true;
+          }
         }
-      );
+      });
     } else {
-      this.props.project.writeJSON(component.src, component.props);
+      try {
+        await this.props.project.writeJSON(component.src, component.props);
+      } catch (e) {
+        console.error(e);
+        this.showDialog(ErrorDialog, {
+          title: "Error Saving Material",
+          message: e.message || "There was an error when saving the material."
+        });
+      }
     }
   };
 
   onLoadComponent = component => {
-    this.props.openFileDialog(
-      async src => {
-        component.src = src;
-        component.shouldSave = true;
-        component.modified = false;
-        component.constructor.inflate(this.state.object, await this.props.project.readJSON(component.src));
-        this.props.editor.signals.objectChanged.dispatch(this.state.object);
-      },
-      {
-        filters: [component.fileExtension],
-        title: "Load material...",
-        confirmButtonLabel: "Load"
+    this.props.showDialog(FileDialog, {
+      filters: [component.fileExtension],
+      title: "Load material...",
+      confirmButtonLabel: "Load",
+      onConfirm: async src => {
+        let loaded = false;
+
+        try {
+          setTimeout(() => {
+            if (loaded) return;
+            this.showDialog(ProgressDialog, {
+              title: "Loading Material",
+              message: "Loading material..."
+            });
+          }, 100);
+
+          component.src = src;
+          component.shouldSave = true;
+          component.modified = false;
+          component.constructor.inflate(this.state.object, await this.props.project.readJSON(component.src));
+          this.props.editor.signals.objectChanged.dispatch(this.state.object);
+          this.hideDialog();
+        } catch (e) {
+          console.error(e);
+          this.showDialog(ErrorDialog, {
```

Comment:
This is bugged. All of the above `this.hideDialog()` calls and `this.showDialog()` calls should be `this.props.hide/showDialog()`